src/main/java/com/example/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,145 @@
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    public List<OrderDto> findAllOrdersByUser(String email, Pageable pageable) {
+        return orderRepository.findAllOrderByUserId(getUserByEmail(email).getId(), pageable)
+                .stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto getOrderItemById(String email, Long orderId, Long itemId) {
+        return getOrderByUserId(email, orderId)
+                .getOrderItems()
+                .stream()
+                .filter(orderItem -> orderItem.getId().equals(itemId))
+                .findFirst()
+                .map(orderItemMapper::toDto).orElseThrow(
+                        () -> new EntityNotFoundException("Can't find item with id: " + itemId)
+                );
+    }
+
+    @Override
+    public OrderDto createNewOrder(String email, CreateOrderRequestDto dto) {
+        User user = getUserByEmail(email);
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserWithItems(user.getEmail())
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Can't find shopping cart by user id: "
+                                + user.getId()));
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Status.PENDING);
+        order.setTotal(countTotal(shoppingCart));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(dto.shippingAddress());
+        Set<OrderItem> orderItems = createOrderItems(order,shoppingCart.getCartItems());
+        order.setOrderItems(orderItems);
+        orderRepository.save(order);
+        orderItemRepository.saveAll(orderItems);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderItemDto> findAllOrderItemsByOrderId(String email, Long orderId) {
+        Order order = getOrderByUserId(email, orderId);
+        return orderItemRepository.findAllByOrder(order)
+                .stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderDto updateOrderStatus(Long orderId, StatusDto dto) {
+        Order order = orderRepository.findById(orderId).orElseThrow(
+                () -> new EntityNotFoundException("Can't find order with id: " + orderId)
+        );
+        if (containsValidStatus(dto.status())) {
+            order.setStatus(dto.status());
+            return orderMapper.toDto(orderRepository.save(order));
+        } else {
+            throw new RuntimeException("Invalid status value");
+        }
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        orderRepository.deleteById(id);
+    }
+
+    private boolean containsValidStatus(Status status) {
```

Comment:
I don't see any point in that validation, StatusDto already has a field that is typed as an enum, which just won't be able to parse something else into it, so it will always be a valid status enum here and we can remove it

Reply:
still present, am I missing something?